webgpu.h

```diff
@@ -1354,10 +1353,10 @@ typedef void (*WGPUProcShaderModuleSetLabel)(WGPUShaderModule shaderModule, char
 
 // Procs of Surface
 typedef WGPUTextureFormat (*WGPUProcSurfaceGetPreferredFormat)(WGPUSurface surface, WGPUAdapter adapter);
-
-// Procs of SwapChain
-typedef WGPUTextureView (*WGPUProcSwapChainGetCurrentTextureView)(WGPUSwapChain swapChain);
-typedef void (*WGPUProcSwapChainPresent)(WGPUSwapChain swapChain);
+typedef void (*WGPUProcSurfaceConfigure)(WGPUSurface surface, WGPUDevice device, WGPUSurfaceConfiguration const * configuration);
+typedef void (*WGPUProcSurfaceUnconfigure)(WGPUSurface surface);
+typedef WGPUTexture (*WGPUProcGetCurrentTexture)(WGPUSurface surface, bool * suboptimal);
+typedef void (*WGPUProcSurfacePresent)(WGPUSurface surface);
```

Comment:
There are other changes @cwfitzgerald made compared to the Web API for getting the supported formats and present mode. Maybe we should replicate some of them, and also re-discuss them in the context of a native API that's shared between projects.

Reply:
#163 mentions `wgpuSurfaceGetSupportedPresentModes`.

for supported formats I think we can have
 ```c
size_t wgpuSurfaceGetSupportedFormats(WGPUSurface surface, WGPUTextureFormats * formats);
```

Reply:
don't think `wgpu` [provides a way to know that](https://docs.rs/wgpu-core/latest/wgpu_core/hub/struct.Global.html#method.surface_get_supported_formats)

(also we don't have hdr support in `webgpu.h`, yet, [wgpu does though](https://docs.rs/wgpu-types/0.13.2/wgpu_types/enum.AstcChannel.html#variant.Hdr))

Reply:
`AstcChannel` seems unrelated to the swapchain API TBH.

Reply:
Ah yes, apologies!

Reply:
Let's tackle this in another PR - maybe after a mini issue-rfc kind of deal - as there is a ton of complexity there that we aren't yet equipped to deal with here. For reference: https://github.com/gfx-rs/wgpu/issues/2869 is our proposal internal to wgpu which would set the stage for the HDR api.